test/integration/good/compiler-optimizations/cpp.expected

```diff
@@ -941,23 +941,33 @@ class ad_level_deep_dependence_model final : public model_base_crtp<ad_level_dee
         Eigen::Matrix<local_scalar_t__, -1, -1>> X_tp1;
       current_statement__ = 2;
       stan::model::assign(X_tp1, X_d, "assigning variable X_tp1");
-      Eigen::Matrix<double, -1, -1> X_tp2;
+      Eigen::Matrix<double, -1, -1> X_tp2 =
+         Eigen::Matrix<double, -1, -1>::Constant(10, 10,
+           std::numeric_limits<double>::quiet_NaN());
       stan::model::assign(lcm_sym9__, stan::math::exp(X_d),
         "assigning variable lcm_sym9__");
       stan::model::assign(X_tp2, lcm_sym9__, "assigning variable X_tp2");
-      Eigen::Matrix<double, -1, -1> X_tp3;
+      Eigen::Matrix<double, -1, -1> X_tp3 =
```

Comment:
These also seem like they're wrong re-introductions of the initializer. Do we have no way around this temporary here? The original model doesn't feature control flow here

Reply:
I thought so at first as well but if you look at the next lines 

```
      stan::model::assign(lcm_sym10__, stan::math::exp(lcm_sym9__),
        "assigning variable lcm_sym10__");
      stan::model::assign(X_tp3, lcm_sym10__, "assigning variable X_tp3");
```

The lazy code motion assigns `lcm_sym10__` right below this and then assigns `X_tp3`. That fails the check so NA's get filled here. We could run this before lazy code motion but I'm not sure if things can go from needing to be initalized to not initialized and vice versa during the different propogations and lcm.

Reply:
So previously it was running earlier?

Reply:
Before the scheme was, once we hit a decl, to iterate though the remaining statements and as long as we see a full assignment before it was used then we would allow it to be not initialized. But the logic is simpler now that we just look at the statement right below the decl. If it is not an assignment then we force it to fill with NA values. It's lazier but it's much safer. imo I like this simpler approach because it hits the bigger misses right now. For instance it stops the params from being filled with NA values (since they are always filled through the deserializer)